### DISCUSSION TITLE
Workflow should show stage in list view and article edit details.

### DISCUSSION THREAD

**Discussion by stimpsonjcat** [joomla/joomla-cms#47098] [6. Duplicates / Archived / Not for discussion]
### Is your feature request related to a problem? Please describe.

There's one thing that's not working right with workflows: neither the author, nor the editor (or any admin) can see the stage of the workflow. When you open the article for editing, the drop down menu of the workflow is showing the title of the workflow, not the stage of the workflow. Also in the list view of articles, the stage column doesn't show the actual stage.
An editor / admin thus doesn't know wether an article is still in draft or ready to be checked & published.

### Describe the solution you'd like

In the list view of articles in the admin and in the edit view of an article, in the admin _and_ in the front end, one should be able to see the stage of the workflow in witch the articles is in.

### Additional context

On one of my sites, where we have a team of authors and editors we use this great feature. Until now we've used the standard workflow that comes with Joomla. That is, only editors can change the state of a workflow and article.

I want to add two new stages to the workflow: draft and ready for publishing. When an author writes an article the state should be draft by default. If the author finishes the article, he can set the workflow to Ready for publishing. At that point the editors should receive a message that there's a new article that needs to be checked and published.

@crystalenka

# Maintainer Information
Unrelated: 18.03.2026
Reason: Already improved in [PR](https://github.com/joomla/joomla-cms/pull/39325) mentioned in [Comment](https://github.com/joomla/joomla-cms/discussions/47098#discussioncomment-15883451)

**Comment by crystalenka** [comment]
Thank you! Can you add the screenshots also that you shared on Facebook? I am on mobile and can't do it currently. 

I'm not sure yet if this is a technical bug or something that was missed when the component was introduced, but will take a look :)

**Comment by brianteeman** [comment]
I think you are witnessing the ui of com_workflows


What do you see when you clcik on the stage name like bellow

![image](https://user-images.githubusercontent.com/1296369/203148046-0922e2de-178d-40b5-b7ff-cf073dbb06f6.png)

**Comment by stimpsonjcat** [comment]
I see a menu with my workflow stages in the list view. The title of the workflow is selected, not the actual stage.

<img width="756" alt="Schermafbeelding 2022-11-21 om 21 05 27" src="https://user-images.githubusercontent.com/3033162/203148531-00f77b20-8d91-42eb-86e7-1c178ffd14de.png">

**Comment by stimpsonjcat** [comment]
This it what it looks like
<img width="1384" alt="Schermafbeelding 2022-11-21 om 20 36 21" src="https://user-images.githubusercontent.com/3033162/203148968-8496d567-be08-4b77-9204-72e861d8dd7a.png">

This is what I would propose.
<img width="1384" alt="Schermafbeelding 2022-11-21 om 20 39 04" src="https://user-images.githubusercontent.com/3033162/203148977-5a1311ce-b137-4147-b9d9-543b26d80f5a.png">

**Comment by brianteeman** [comment]
Exactly what I thought. The list is a list of transitions that are valid for the current stage. The currennt stage is the selected item.

Confused? Yes I am sure you are.

**Comment by crystalenka** [comment]
It's not clear at all what stage an item might be in until you click, and only maybe at that point. 

I asked him to file the ticket, @brianteeman , I am sure this can be improved.

**Comment by brianteeman** [comment]
@crystalenka I have been saying this before workflows were even merged. :( The problem is that stage != status and transition != anything. You can only use workflows if you forget everything you already know.

@stimpsonjcat I don't know if it will help but I gave a tutorial on workflows at joomladay fr earlier this year. I really tried to simplify things - it was the only way I could understand it.

https://youtu.be/oX70QImF5WY

@crystalenka I filed multiple bug reports while preparing that tutorial. Pretty sure they are all still open.

**Comment by crystalenka** [comment]
@brianteeman I will take a look from my computer to see if I can find those issues. :) I remember discussing with you the various UX issues with workflows. 

JXT has a lot on our plate right now since we're trying to catch up with everything, but workflows are on the list somewhere....

**Comment by Rolli1962** [comment]
In my opinion the workflow is a very good feature. It has only to be understood. Maybe it would be a little bit easier if in the example after installation more than one stage would be given.
Have created a workflow what think [stimpsonjcat](https://github.com/stimpsonjcat) is thinking of - with that different stages. In the back-end the view is correct. Means I think there are no coding issues:
![wf-testview](https://user-images.githubusercontent.com/60449692/204057614-cd8ac443-72a6-4294-adb2-44e941f8b427.jpg)

**Comment by chmst** [comment]
In the list view the stage is visible.  

It depends on your own user group what you can see. As a super user you see all articles in all stages. As user you see only the articles where you have access. So it is possible that a user sees always the same stage can perform only one action for exactly one transition. 

Stage and state of articles are visible in the "published" area in backend and frontend. 
![grafik](https://user-images.githubusercontent.com/1035262/204082760-8e3f8a37-148a-4b69-8855-68af68a4282c.png)

**Comment by chmst** [comment]
@stimpsonjcat
The workflow concept is somehow difficult to understand and the best is learning by doing. 
Could we explain how workflow works? Do you still think that this is an issue?

**Comment by Rolli1962** [comment]
My understanding of the question of this thread is make the status (stage) of articles visible in an overview. If the workflow-stages are configured / used in a right way, the stages are visible in the back-end. But even in the current J4.2.5 version this is not in the frontend (see example, same entries like listed before in the back-end). Or can I learn and have to do some settings ?
![wf-testview-frontend](https://user-images.githubusercontent.com/60449692/204090859-9d7e704e-ac4e-44e5-baa1-7dc458bcaaf9.jpg)

**Comment by brianteeman** [comment]
I nfound that the only way to be able to understand it is to not use the supplied workflow and to create your own from scratch. Only then do the confusing terms make sense. It took me many hours to even begin to understand it

**Comment by chmst** [comment]
same here :) As I said, learning by doing - and still have problems with "stage" and "state".

**Comment by Rolli1962** [comment]
> @stimpsonjcat The workflow concept is somehow difficult to understand and the best is learning by doing. Could we explain how workflow works? Do you still think that this is an issue?
To be honest I had also to learn how it works. There are quite good explanations in the internet. But think like me first step is simple to try the given example in the installation. There is only one stage with different transitions. It takes a while until the understanding the whole thing makes only sense with more stages than one. Maybe as short-term solution it makes sense to set a link to the existing detailed descriptions on the internet?

**Comment by stimpsonjcat** [comment]
Ah, I'm getting it. I've been confusing stages with transitions. So the article (list) view shows the stage, not the transition. Though you can set a transition from the list view directly. The dropdown menu will show the the current stage as a title and the transitions (depending on your rights) below that.
I really love the workflow tool. It gives Joomla a clear advantage over other cms-es. 👍

**Comment by Rolli1962** [comment]
Seems I am to stupid to create an account on docs.joomla.org - Maybe another one has an account there?
Here a proposal for a 2nd scenario for workflows. Hope it matches as least as basis for own creation. 
And of course I am also to stupid to create here a readable content - as hidden article on one of my sites:
[possible workflow-scenario-2 for docs](https://dk4rh.de/index.php/hidden/workflow-scenario-2)

**Comment by crystalenka** [comment]
You are not too stupid, it's just really really hard. The captcha is impossible. I gave up, too.

**Comment by stimpsonjcat** [comment]
I think it would be easier to understand for users / admins if the default workflow that comes with Joomla 4 would have more than one stage. The one stage now has transitions that seem to be no more than one click to (un)publish and (un)feature an article. I would suggest to create an extra 'frontend' stage for authors to set an article to draft and allowing the author to set the article to something like 'check before publishing'.

**Comment by chmst** [comment]
Thanks for your suggestion. I add the ne feature label so that it will be discussed for later versions.

**Comment by brianteeman** [comment]
Please look at the changes proposed in #39325 as I believe it removes some of the ui confusions
